scala-package/examples/src/main/java/org/apache/mxnetexamples/infer/javapi/objectdetector/SSDClassifierExample.java

```diff
@@ -0,0 +1,201 @@
+		return objDet.imageObjectDetect(img, 3);
+	}
+	
+	static List<List<List<ImmutablePair<String, List<Float>>>>>
+	runObjectDetectionBatch(String modelPathPrefix, String inputImageDir, List<Context> context) {
+		Shape inputShape = new Shape(new int[]{1, 3, 512, 512});
+		List<DataDesc> inputDescriptors = new ArrayList<DataDesc>();
+		inputDescriptors.add(new DataDesc("data", inputShape, DType.Float32(), "NCHW"));
+		ObjectDetector objDet = new ObjectDetector(modelPathPrefix, inputDescriptors, context, 0);
+		
+		// Loading batch of images from the directory path
+		List<List<String>> batchFiles = generateBatches(inputImageDir, 20);
+		List<List<List<ImmutablePair<String, List<Float>>>>> outputList
+				= new ArrayList<List<List<ImmutablePair<String, List<Float>>>>>();
+		
+		for (List<String> batchFile : batchFiles) {
+			List<BufferedImage> imgList = ObjectDetector.loadInputBatch(batchFile);
+			// Running inference on batch of images loaded in previous step
+			List<List<ImmutablePair<String, List<Float>>>> tmp
+					= objDet.imageBatchObjectDetect(imgList, 5);
+			outputList.add(tmp);
+		}
+		return outputList;
+	}
+	
+	static List<List<String>> generateBatches(String inputImageDirPath, int batchSize) {
+		File dir = new File(inputImageDirPath);
+
+		List<List<String>> output = new ArrayList<List<String>>();
+		List<String> batch = new ArrayList<String>();
+		for (File imgFile : dir.listFiles()) {
+			batch.add(imgFile.getPath());
+			if (batch.size() == batchSize) {
+				output.add(batch);
+				batch = new ArrayList<String>();
+			}
+		}
+		if (batch.size() > 0) {
+			output.add(batch);
+		}
+		return output;
+	}
+	
+	public static void main(String[] args) {
+		SSDClassifierExample inst = new SSDClassifierExample();
+		CmdLineParser parser  = new CmdLineParser(inst);
+		try {
+			parser.parseArgument(args);
+		} catch (Exception e) {
+			logger.error(e.getMessage(), e);
+			parser.printUsage(System.err);
+			System.exit(1);
+		}
+		
+		String mdprefixDir = inst.modelPathPrefix;
+		String imgPath = inst.inputImagePath;
+		String imgDir = inst.inputImageDir;
+		
+		if (!checkExist(Arrays.asList(mdprefixDir + "-symbol.json", imgDir, imgPath))) {
+			logger.error("Model or input image path does not exist");
+			System.exit(1);
+		}
+		
+		List<Context> context = new ArrayList<Context>();
+		if (System.getenv().containsKey("SCALA_TEST_ON_GPU") &&
+				Integer.valueOf(System.getenv("SCALA_TEST_ON_GPU")) == 1) {
+			context.add(Context.gpu());
+		} else {
+			context.add(Context.cpu());
+		}
+		
+		try {
+			Shape inputShape = new Shape(new int[] {1, 3, 512, 512});
+			Shape outputShape = new Shape(new int[] {1, 6132, 6});
+			
+			
+			int width = inputShape.get(2);
+			int height = inputShape.get(3);
+			String outputStr = "\n";
+			
+			List<List<ImmutablePair<String, List<Float>>>> output
+					= runObjectDetectionSingle(mdprefixDir, imgPath, context);
+			
+			for (List<ImmutablePair<String, List<Float>>> ele : output) {
```

Comment:
These complex nested structures make it a bit hard to read. Going by @nswamy 's suggestion, a POJO will definitely help ease out the readability.

Reply:
Yeah I agree. Already started writing it.